app/src/main/java/org/openobservatory/ooniprobe/common/CountlyManager.java

```diff
@@ -0,0 +1,71 @@
+package org.openobservatory.ooniprobe.common;
+
+import android.content.Context;
+
+import com.google.android.gms.common.util.ArrayUtils;
+
+import org.openobservatory.engine.Engine;
+import org.openobservatory.ooniprobe.BuildConfig;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import ly.count.android.sdk.Countly;
+import ly.count.android.sdk.CountlyConfig;
+import ly.count.android.sdk.DeviceId;
+import ly.count.android.sdk.messaging.CountlyPush;
+
+public class CountlyManager {
+    private static String[] basicFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.location,
+    };
+
+    private static String[] analyticsFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.sessions,
+            ly.count.android.sdk.Countly.CountlyFeatureNames.views,
+            ly.count.android.sdk.Countly.CountlyFeatureNames.events
+            //TODO evaluate scrolls, clicks, forms, attribution
+    };
+
+    private static String[] crashFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.crashes,
+    };
+
+    private static String[] pushFeatures = new String[]{
+            ly.count.android.sdk.Countly.CountlyFeatureNames.push,
+    };
+
+    public static void register(Context ctx, PreferenceManager preferenceManager){
+        CountlyConfig config = new CountlyConfig()
+                .setAppKey("146836f41172f9e3287cab6f2cc347de3f5ddf3b")
+                .setContext(ctx)
+                .setDeviceId(Engine.getUUID())
```

Comment:
I think what you want here is to get the UUID from the configuration and insert a new UUID into the configuration only if it has not already been inserted. The current code uses a new identity every time.